### PR TITLE
Fix typo in `instance` usage example

### PR DESCRIPTION
**Pull request by Morriar**
I believe what we want an `Array[instance]`?

### CODE REVIEW

File: docs/syntax.md [hunk 1]
```diff
@@ -291,7 +291,7 @@ class Foo
 
   def foo: () -> instance
```
Comment (Morriar) [inline docs/syntax.md:292]: @soutaro I'm also surprised by how it's used in this case? What's the meaning of it? Is it the same as `self`?

Comment (ParadoxV5) [reply docs/syntax.md:292]: Dummy names like ‘foo’ don’t always make comprehensible general examples. 👍

Comment (soutaro) [reply docs/syntax.md:292]: @Morriar This is in fact same with `self`. They are different on `def self?.foo: `, but yeah, it's exactly same in this case.

**Review by soutaro — APPROVED** [review]
🙏

**Pull request merged**